src/hooks/usePairVolume24hUSD.ts

```diff
@@ -6,52 +6,142 @@ import Decimal from 'decimal.js-light'
 import { DateTime } from 'luxon'
 import { useMemo } from 'react'
 
+import { subgraphBlocksClients } from '../apollo/client'
 import { ZERO_USD } from '../constants'
+import { useActiveWeb3React } from '../hooks'
+import { SWPRSupportedChains } from '../utils/chainSupportsSWPR'
 
-const QUERY = gql`
-  query getPair24hVolume($pairOrTokenAddress: ID!, $date: Int!) {
-    pairDayDatas(first: 1, where: { pairAddress: $pairOrTokenAddress, date: $date }) {
-      dailyVolumeUSD
+const GET_BLOCK_BY_TIMESTAMP = gql`
```

Comment:
how many more queries are we going to have? At what point should we make a dict out of these?

Reply:
a map, a {key, value} object.

Reply:
I think we could look into that, maybe create an issue?  I think it's out of scope of this PR.